pkg/logs/processor/processor.go

```diff
@@ -58,6 +60,11 @@ func (p *Processor) run() {
 			metrics.LogsProcessed.Add(1)
 			metrics.TlmLogsProcessed.Inc()
 
+			select {
```

Comment:
Under very high load this can be lossy. 
Since multiple pipelines funnel into the same diagnosticChannel I don't want to risk slowing down/blocking the pipelines. Thoughts?